HIP/hip-868.md

```diff
@@ -0,0 +1,155 @@
+expect SELFDESTRUCT to operate as a sweep in some cases. Outside the EVM,
+however, there is no need to change any semantics. HAPI calls can still be used
+to delete a contract and state expiry can still delete an account.
+
+## Rationale
+
+Hedera will support the [EIP-6780](https://eips.ethereum.org/EIPS/eip-6780)
+change to the SELFDESTRUCT operation. If the contract existed prior to the
+transaction the contract will not be destroyed and the contents will instead be
+swept to the beneficiary. Contracts within the same transaction will be
+destroyed as per prior behavior.
+
+However, since Hedera places "ownership" of token balances with the account
+holding the balance we will need to extend the behavior of the sweep mode to
+also transfer all HTS tokens to the beneficiary. Notably, for a successful
+transaction beneficiary accounts must support the receiving of HTS tokens via
+association configuration. This is consistent with current self-destruct
+behavior and consistent with the spirit of the Ethereum change in that code and
+the account are no longer deleted.
+
+When the self-destruct is invoked on a contract within a transaction the
+behavior of Hedera will remain as it currently is in all respects.
+
+## User stories
+
+* As a smart contract developer I want to use the "sweep" semantics of EIP-6780
+  in my contracts to sweep out value (HBAR and HTS token balance) while not
+  destroying the contract.
+* As a smart contract developer I want to be able to continue to use the
+  transient contract pattern and have it behave the same way prior to this HIP.
+* As a HAPI user I want to be able to control my contract account in the same
+  ways I did prior to this HIP.
+
+## Specification
+
+There are no changes needed to the JSON-RPC relay.
+
+### EVM Implementation
+
+The SelfDestructOperation will be re-implemented with hedera semantics. An
+existing hedera version of self-destruct exists that may be manipulated in a way
+similar to how the Besu EVM's implementation was altered.
+
+The existing transient self-destruct behaviors will be retained, and the new
+sweep mode changes will be added. All hBar and HTS tokens will be swept as part
+of the self-destruct to the beneficiary address, what will change is that the
+contract will not be marked for removal.
+
+### Hedera Services
+
+Hedera services may need to be changed to ensure that the transaction processing
+does not leave a "shadow" account number for transient self-destructs. Such
+transient self-destructs will also need to be tested to ensure all HTS tokens
+are properly swept during the sweep as well as the transient contract modes.
+
+## Backwards Compatibility
+
+This change could break contracts that depend on selfdestruct destroying old
+contracts. Relying on such behaviors has been discouraged for years and formally
+deprecated for at least a year on Ethereum Mainnet, so it is expected that such
+use cases have never found their way onto Hedera's Mainnet.
```

Comment:
Is it worth looking at mirror node data to see if in fact it _has_ happened on mainnet? (Or confirm it hasn't, as expected.)

Reply:
Yes, let's check